src/App.tsx

```diff
@@ -1,26 +1,197 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { USER_ID } from './api/todos';
+import { Header } from './components/Header';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorMessage } from './components/ErrorMessage';
+import { Todo } from './types/Todo';
+import * as todoService from './api/todos';
+import { Options } from './types/Options';
+import { Errors } from './types/Errors';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [errorMessage, setErrorMessage] = useState<Errors | null>(null);
+  const [selectedOption, setSelectedOption] = useState<Options>(Options.All);
+  const [focus, setFocus] = useState(false);
+
+  const filterTodos = (posts: Todo[], option: Options) => {
+    switch (option) {
+      case Options.Active:
+        return posts.filter(post => !post.completed);
+
+      case Options.Completed:
+        return posts.filter(post => post.completed);
+
+      default:
+        return posts;
+    }
+  };
+
+  const todoFilterSelectedOptions = filterTodos(todos, selectedOption);
+  const todoFilterOptionActive = filterTodos(todos, Options.Active);
+
+  const clearError = useCallback(() => {
+    setErrorMessage(null);
+  }, []);
+
+  const showError = useCallback(
+    (error: Errors) => {
+      setErrorMessage(error);
+
+      setTimeout(() => {
+        clearError();
+      }, 3000);
+    },
+    [clearError],
+  );
+
+  useEffect(() => {
+    setFocus(true);
+
+    todoService
+      .getTodos()
+      .then(setTodos)
+      .catch(() => showError(Errors.LoadTodos));
+  }, [showError]);
+
+  const addTodo = useCallback(
+    async ({ userId, title, completed }: Omit<Todo, 'id'>) => {
+      const newTempTodo: Todo = {
+        id: 0,
+        userId,
+        title,
+        completed,
+      };
+
+      setTempTodo(newTempTodo);
+
+      try {
+        const newTodo = await todoService.createTodos({
```

Comment:
```suggestion
        const newTodo = await todoService.createTodo({
```
it works with one todo

fix similar cases